### ISSUE TITLE
Possible to play as Rash as player 1/controller 1?

### ISSUE THREAD

**Issue by AlWesker** [maxim-zhao/battlemaniacs-restoration#10]
The Master System conversion of Battletoads in Battlemaniacs inherits and aspect which RARE defaulted for both Battletoads for the NES and Battletoads in Battlemaniacs for the SNES. In these both games you can play as one of two possible Toads while the third one is captured and non playable. Battletoads for the NES assigned Rash to Player 1 and Zitz to Player 2, while Battletoads in Battlemaniacs for the SNES assigned Pimple to Player 1 and Rash to Player 2. The Master System conversion respects the SNES procedure, and Battletoads in Battlemaniacs actually has different looking graphics/animations for each Toad which make them distinctive each other (while Battletoads for the NES relied on the same but color swapped sprites) adding to gameplay. So in a 2 player game, Player 1 will always be Pimple and Player 2 will always be Rash. But in a 1 player game, if you play with the first controller you always will be Pimple, and if you want to play as Rash in a 1 player game then you have to use the second controller or switch ports. That's easy to do either on a Master System or on a Mega Drive, but if you play the game on an unmodified Game Gear you'll find yourself stuck to play the game as Pimple. I don't know if it would be possible to add a simple workaround to play as Rash on a 1 player game with the first controller. Maybe it would be possible to add something like, if you start the 1 player game with button 1 you'll be Pimple while if you start it with button 2 you would be Rash. This would be a simple way which wouldn't alter too much the other aspects or the game (adding a proper player select screen would surely be huge work) but is it that simple to implement?

**Comment by maxim-zhao** [comment]
A player select screen is not out of the question - consider that inserting screens is easier than modifying existing ones.

**Comment by maxim-zhao** [comment]
I had a look at the code, as might be expected the code is tying the Rash character to the player 2 inputs, so it's a bit hard to separate them.

**Comment by AlWesker** [comment]
Shame. Oh well, I suppose the game will have to remain locked to Pimple if you play on a Game Gear or similar system.

**Comment by PyronsLair** [comment]
Rash sprite looks amazing on sms, i also vote to try go deep on this..
i have no doubts that everyones gonna enjoy select your favorite toad..
]its a nice feature and improve the game over original

**Comment by AlWesker** [comment]
Yeah, it is. The fact is, if you are playing this on a Master System or a Mega Drive or emulator or any other compatible/emulator system that allows you to use a second controller, then it's easy to get either the second controller or switch the controller on the first port to the second port and play as Rash with it. It was also that way on the SNES version and they decided to mimick that on the Master System version.

The problem is if you are playing this on a Game Gear. Unless you modify the hardware to allow for a second controller port, it's impossible to play as Rash there. You are stuck to play as Pimple since he's forced to be the first controller character. But even if you are playing on a modified Game Gear with a second controller port (or a Nomad, which already has one) it's unfitting because you are playing on a portable system using a controller instead of the built in controls just to play as Rash. 

 So this was proposed in order to allow you play as Rash with the first controller so it may fix the inconvenient on those systems, but it seems it's difficult to do. Oh well.

**Comment by maxim-zhao** [comment]
I'd rather leave it for later, it's not a must-have for me - and it's hard!
